src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -1068,6 +1068,12 @@ CoordinatedRemoteTransactionsSavepointRollback(SubTransactionId subId)
 		MultiConnection *connection = dlist_container(MultiConnection, transactionNode,
 													  iter.cur);
 		RemoteTransaction *transaction = &connection->remoteTransaction;
+
+		/* cancel any ongoing queries before issuing rollback */
+		ClearResultsIfReady(connection);
```

Comment:
Probably not necessary to do this before the cancellation. The cancellation actually goes over a separate connection.

Reply:
removed.